Base/QTCore/qSlicerExtensionsManagerModel.cxx

```diff
@@ -2038,35 +1999,47 @@ bool qSlicerExtensionsManagerModel::updateExtensionsMetadataFromServer(bool forc
         return false;
         }
       d->ExtensionsMetadataFromServerAPI.setServerUrl(this->serverUrl().toString() + QString("/api/v1/app/%1/extension").arg(appID));
+      parameters["app_revision"] = this->slicerRevision();
+      parameters["os"] = this->slicerOs();
+      parameters["arch"] = this->slicerArch();
+      // request all metadata in a single response (it makes synchronous query simpler)
+      parameters["limit"] = QString::number(0);
```

Comment:
This makes sense, I was initially setting the value to `-1`

> limit (optional): the maximum number of results to return. A limit of 0 (the default) is equivalent to setting no limit.

Source: https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html#pymongo.collection.Collection.find

Reply:
I had to set the limit to 0 because setting it to -1 makes it return a single item: https://slicer-packages.kitware.com/api/v1/app/5f4474d0e1d8c75dfc705482/extension?app_revision=30338&os=win&limit=-1

Reply:
And since the json used for testing only include one item, I initially missed that.

Reply:
This is also confirmed using either:
* https://slicer-packages.kitware.com/api/v1/app/5f4474d0e1d8c75dfc705482/extension?os=linux&arch=amd64&app_revision=31036&limit=0
* https://slicer-packages.kitware.com/api/v1/app/5f4474d0e1d8c75dfc705482/extension?os=linux&arch=amd64&app_revision=31036&limit=-1